frontends/web/src/routes/account/summary/chart.tsx

```diff
@@ -15,157 +15,215 @@
  */
 
 import { createChart, IChartApi, LineData, LineStyle, LogicalRange, ISeriesApi, UTCTimestamp, MouseEventParams, ColorType, Time } from 'lightweight-charts';
-import { Component, createRef, ReactChild } from 'react';
-import { ISummary } from '../../../api/account';
-import { translate, TranslateProps } from '../../../decorators/translate';
+import { useContext, useEffect, useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { usePrevious } from '../../../hooks/previous';
+import { ISummary, ChartData } from '../../../api/account';
 import { Skeleton } from '../../../components/skeleton/skeleton';
 import { Amount } from '../../../components/amount/amount';
 import { PercentageDiff } from './percentage-diff';
 import { Filters } from './filters';
 import { getDarkmode } from '../../../components/darkmode/darkmode';
-import { TChartFiltersProps } from './types';
 import { DefaultCurrencyRotator } from '../../../components/rates/rates';
 import { AppContext } from '../../../contexts/AppContext';
 import styles from './chart.module.css';
 
-export interface FormattedLineData extends LineData {
-  formattedValue: string;
-}
-
-export type ChartData = FormattedLineData[];
-
-type ChartProps = {
-  data: ISummary;
-  noDataPlaceholder?: ReactChild;
-  hideAmounts: boolean;
+type TProps = {
+  data?: ISummary;
+  noDataPlaceholder?: JSX.Element;
+  hideAmounts?: boolean;
 };
 
-type State = {
-  source: 'daily' | 'hourly';
-  difference?: number;
-  diffSince?: string;
-  toolTipVisible: boolean;
-  toolTipValue?: string;
-  toolTipTop: number;
-  toolTipLeft: number;
-  toolTipTime: number;
-  isMobile: boolean;
-}
-
-type Props = ChartProps & TranslateProps;
+const defaultData: Readonly<ISummary> = {
+  chartDataMissing: true,
+  chartDataDaily: [],
+  chartDataHourly: [],
+  chartFiat: 'USD',
+  chartTotal: null,
+  formattedChartTotal: null,
+  chartIsUpToDate: false,
+  lastTimestamp: 0,
+};
 
 type FormattedData = {
   [key: number]: string;
 }
-class Chart extends Component<Props, State> {
-  static contextType = AppContext;
-  context!: React.ContextType<typeof AppContext>;
-  private ref = createRef<HTMLDivElement>();
-  private refToolTip = createRef<HTMLSpanElement>();
-  private chart?: IChartApi;
-  private lineSeries?: ISeriesApi<'Area'>;
-  private height: number = 300;
-  private mobileHeight: number = 150;
-  private formattedData?: FormattedData;
-
-  static readonly defaultProps: ChartProps = {
-    data: {
-      chartDataMissing: true,
-      chartDataDaily: [],
-      chartDataHourly: [],
-      chartFiat: 'USD',
-      chartTotal: null,
-      formattedChartTotal: null,
-      chartIsUpToDate: false,
-      lastTimestamp: 0,
-    },
-    hideAmounts: false,
+
+function getUTCRange() {
```

Comment:
please use arrow functions if possible, I recently changed other functions see https://github.com/BitBoxSwiss/bitbox-wallet-app/pull/2731

Reply:
Good to know, so I'll always use them now. I was really confused on how to decide which ones to use..I kept the normal functions so the `git diff` is better because they don't need to be moved around (hoisting)

Reply:
actually typescript did not warn me when I changed this, had to move the functions around quite a bit so the order is correct and I did not get any runtime errors...